pkg/agent/bgp/gobgp/gobgp.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gobgp
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+	"time"
+
+	gobgpapi "github.com/osrg/gobgp/v3/api"
+	"github.com/osrg/gobgp/v3/pkg/server"
+	"google.golang.org/protobuf/types/known/anypb"
+	"k8s.io/utils/net"
+
+	"antrea.io/antrea/pkg/agent/bgp"
+)
+
+const (
+	ipv4AllZero = "0.0.0.0"
+	ipv6AllZero = "::"
+)
+
+type Server struct {
+	server       *server.BgpServer
+	globalConfig *gobgpapi.Global
+}
+
+func NewGoBGPServer(globalConfig *bgp.GlobalConfig) bgp.Interface {
```

Comment:
My understand is that go recommends running the actual type here ``* Server`, not the interface

> Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring.

https://go.dev/wiki/CodeReviewComments

Reply:
Done. Thanks for pointing out this. Will follow the practice.